src/storage/src/source/mysql/timestamp.rs

```diff
@@ -77,23 +94,23 @@ impl SourceTimestamp for TransactionId {
             PartitionId::None,
             "invalid non-partitioned partition {pid}"
         );
-        let id: i64 = offset.offset.try_into().expect("invalid transaction id");
+        let id: u64 = offset.offset;
         Self::new(id)
     }
 
     fn try_into_compat_ts(&self) -> Option<(PartitionId, MzOffset)> {
-        let id: u64 = self.0.try_into().expect("verified non-negative");
+        let id: u64 = self.0;
         Some((PartitionId::None, MzOffset::from(id)))
     }
 
     fn encode_row(&self) -> Row {
-        Row::pack([Datum::Int64(self.0)])
+        Row::pack([Datum::UInt64(self.0)])
```

Comment:
I'm torn on what the best choice here is. The reason I had chosen `Int64` here instead of `UInt64` is because `UInt64` is an mz extension to the postgres wire protocol. This is no problem for us, since we trivially decode it but it does mean that any users interacting with Materialize in a programmatic way will need to deal with parsing our extension of the protocol. Postgres and kafka already use `Uint64` so maybe the ship has sailed, I don't know. Just wanted to share the context behind the `i64` choice

Reply:
Small nit: unsigned types are not an extension of the wire protocol itself! Rather, they just present as custom types—ones that you can even add to PostgreSQL with the pguint extension: https://github.com/petere/pguint

Your point stands—PostgreSQL drivers won't have built-in support for the type—but also almost every PostgreSQL driver has relatively straightforward extension points to plug in encoders/decoders for custom types, since they're very common in PostgreSQL itself.

Apologies if I'm being pedantic, but I thought it was worth clarifying the exact amount of extension-ness in case it changed your thinking, Petros.

Reply:
I don't have a super strong opinion here, but I switched this to make it easier to work directly with the values in the mysql-common serialization, which are `Uint64`s https://github.com/blackbeam/rust_mysql_common/blob/master/src/binlog/consts.rs#L719 

However it appears that in mysql itself these are signed int64s https://dev.mysql.com/doc/dev/mysql-server/latest/namespacemysql_1_1gtid.html#a8679e4b2ae8c097668a3ea89b85c54b2 so we would likely be fine with converting to a `Int64` if we needed to.

I talked to @petrosagg offline and for now I'm going to leave this as-is in this PR, but in the next PR to switch to the PartitionedTimestamp I will re-evaluate and we can potentially implement the `UInt64 -> Int64` casting in the `encode_row` and `decode_row` implementations

Reply:
That makes sense to me. I think trying to match the upstream system's native type for the offsets is a good way to go. (Although it does make me wonder why we chose `u64` for Kafka offsets, since I believe those are `i64`s upstream.)